tests/subsAndMods.test.ts

```diff
@@ -0,0 +1,361 @@
+  createMockVolume,
+  START_TIME,
+  EMPTY_CHATTERS,
+  DEFAULT_TEST_SETTINGS,
+  simSetSubscribers,
+  simSetModerators,
+} from "./simulation.js";
+import { Queue, QueueDataMap, OnlineOfflineList } from "../src/queue.js";
+import {
+  EventSubChannelSubscriptionEvent,
+  EventSubChannelSubscriptionEndEvent,
+  EventSubChannelModeratorEvent,
+} from "@twurple/eventsub-base";
+import { QueueSubmitter } from "../src/extensions-api/queue-entry.js";
+
+// Set up the fake timers
+jest.useFakeTimers();
+
+// Basic pre-setup stolen from weight.test.ts
+const setupMocks = () => {
+  // reset chatters
+  simSetChatters(EMPTY_CHATTERS);
+
+  // reset time
+  jest.setSystemTime(START_TIME);
+};
+
+beforeEach(setupMocks);
+
+async function setupQueue() {
+  const volume = createMockVolume();
+  const index = await simRequireIndex(
+    volume,
+    DEFAULT_TEST_SETTINGS,
+    START_TIME,
+    setupMocks
+  );
+  const twitch = index.twitch;
+  const queue: Queue = index.quesoqueue;
+  if (queue.testAccess === undefined) {
+    throw new Error("testAccess is undefined");
+  }
+
+  const getList: QueueDataMap<OnlineOfflineList> = await queue.list();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(0);
+    expect(list.offline).toHaveLength(0);
+  });
+
+  const chatter1 = buildChatter("chatter1", "chatter1", false, false, false);
+  const chatter2 = buildChatter("chatter2", "chatter2", false, false, false);
+  const chatter3 = buildChatter(
+    "chatter3",
+    "chatter3",
+    false,
+    false,
+    false,
+    "chatter3"
+  );
+
+  const level1 = "D36-010-5YF";
+  const level2 = "MY2-H2M-DSG";
+  const level3 = "GBJ-6QY-P8G";
+
+  expect(queue.add(level1, chatter1)).toContain("has been added to the queue");
+  expect(queue.add(level2, chatter2)).toContain("has been added to the queue");
+  expect(queue.add(level3, chatter3)).toContain("has been added to the queue");
+
+  return {
+    queue,
+    chatter1,
+    chatter2,
+    chatter3,
+    twitch,
+  };
+}
+
+test("fetch subs", async () => {
+  const { queue, chatter1, chatter2, chatter3 } = await setupQueue();
+  if (queue.testAccess === undefined) {
+    throw new Error("testAccess is undefined");
+  }
+
+  // Add a subscriber to the "backend"
+  simSetSubscribers([chatter1]);
+
+  // Add our mock users to chat
+  simSetChatters([chatter1, chatter2, chatter3]);
+
+  // Ensure there are no subs yet
+  let getList: QueueDataMap<OnlineOfflineList> = await queue.sublist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(0);
+  });
+
+  // Wait 15 minutes, check for chatter1 as a sub
+  await simAdvanceTime(15 * 60 * 1000);
+
+  // Ensure our sub is now online
+  getList = await queue.sublist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(1);
+    expect(list.online.at(0)?.submitter).toEqual({
+      ...chatter1,
+      isSubscriber: true,
```

Comment:
`submitter` is of type `QueueSubmitter`, because `list.online.at(0)` is of type `QueueEntry` (and nullish). Checking `isSubscriber` is not part of the interface, and I think we should not check this implementation detail here.
Using `sublist` alone checks that the user is a subscriber.
Also if the queue was reloaded from disk, then `isSubscriber` is `undefined`!

Reply:
Switched to just checking the ID, which does the important part: makes sure that the correct user is present.